### PR TITLE
add  the reverse mode of temperature_fan

### PR DESCRIPTION
**Pull request by CF3B5**
add the reverse working mode of temperature_fan to some internal circulation heating fans.

Signed-off-by: Locke chen <cf3b5bb@gmail.com>

### CODE REVIEW

**Comment by Sineos** [comment]
Many thanks for your contribution.
Without judging code or implementation:
* Your contribution needs a sign off line according to https://github.com/Klipper3d/klipper/blob/master/docs/CONTRIBUTING.md
* The setting needs documentation in the applicable sections as well

**Comment by claytono** [comment]
@CF3B5 I've been wanting something exactly like this.  Thanks :)

**Comment by KevinOConnor** [comment]
Thanks.  As a high-level comment, I'm not really sure what this PR does.  As a result, I think it will need a reviewer to help evaluate the benefits.

-Kevin

**Comment by claytono** [comment]
@KevinOConnor I think the intent is to use this to control things like under bed fans.  You could have a set point for the chamber temperature, then use this to turn the fan off once the chamber temperature reaches that set point.  At least that was my hope when I started watching this PR :)

File: docs/Config_Reference.md [hunk 1]
```diff
@@ -2485,6 +2485,10 @@ additional information.
 #gcode_id:
 #   If set, the temperature will be reported in M105 queries using the
 #   given id. The default is to not report the temperature via M105.
+#reverse: False
+#   If true, the working mode of the fan is reversed. If the temperature
+#   is lower than the target temperature, the fan speed increases;
+#   otherwise, the fan speed decreases.
```
Comment (dingyifei) [inline docs/Config_Reference.md:2491]: You need to mention the default is False

**Review by dingyifei — COMMENTED** [review]
Make sure your commit message follow https://www.klipper3d.org/CONTRIBUTING.html#format-of-commit-messages

**Comment by kmobs** [comment]
I'm a big fan of this. Works really nicely for adding a chamber heater as well.

**Comment by Henlor** [comment]
Really nice feature! I'm hoping it would be implemented :)

**Comment by dingyifei** [comment]
Any update on this PR's status?

**Pull request closed without merging**